zcash_primitives/src/transaction/builder.rs

```diff
@@ -306,14 +440,76 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
             .ok_or(Error::InvalidAmount)
     }
 
+    pub fn compute_balances<BalanceError>(
+        &self,
+        fee_strategy: &impl FeeStrategy<Error = BalanceError>,
+    ) -> Result<TransactionBalances, BalanceError>
+    where
+        Error: From<BalanceError>,
+    {
+        fee_strategy.compute_balances(
+            self.transparent_builder.inputs(),
+            self.transparent_builder.outputs(),
+            self.sapling_builder.inputs(),
+            self.sapling_builder.outputs(),
+        )
+    }
+
+    #[cfg(feature = "zfuture")]
+    pub fn compute_balances_zfuture<BalanceError>(
+        &self,
+        fee_strategy: &impl FutureFeeStrategy<Error = BalanceError>,
+    ) -> Result<TransactionBalances, BalanceError>
+    where
+        Error: From<BalanceError>,
+    {
```

Comment:
- Needs documentation.
- Document this addition in the changelog.
- Ditto:

```suggestion
    pub fn compute_balances_zfuture<S: FutureFeeStrategy>(
        &self,
        fee_strategy: &S,
    ) -> Result<TransactionBalances, S::Error> {
```